### PR TITLE
Update setup instructions in the Readme

### PR DESCRIPTION
**Pull request by native-api**
See the rendered README at https://github.com/native-api/pyenv/tree/readme#readme

Make sure you have checked all steps below.

### Prerequisite
* [x] Please consider implementing the feature as a hook script or plugin as a first step.
  * pyenv has some powerful support for plugins and hook scripts. Please refer to [Authoring plugins](https://github.com/pyenv/pyenv/wiki/Authoring-plugins) for details and try to implement it as a plugin if possible.
* [x] Please consider contributing the patch upstream to [rbenv](https://github.com/rbenv/rbenv), since we have borrowed most of the code from that project.
  * We occasionally import the changes from rbenv. In general, you can expect changes made in rbenv will be imported to pyenv too, eventually.
  * Generally speaking, we prefer not to make changes in the core in order to keep compatibility with rbenv.
* [x] My PR addresses the following pyenv issue (if any)
  - Closes https://github.com/pyenv/pyenv/issues/2062
  - Closes https://github.com/pyenv/pyenv/issues/2058
  - Closes https://github.com/pyenv/pyenv/issues/2068
  - Closes https://github.com/pyenv/pyenv/issues/1990

### Description
- [x] Here are some details about my PR

* Update the setup code in the README based on user feedback
* Remove setup code from installer output since it proved

Considered adding logic for pyenv-installer that would replicate the README (like RVM does). Since the logic proves rather complex and multi-case, let's wait for positivbe feedback from users on the README-driven logic before we can rely on it enough to apply it unattended.

**This is my last-ditch effort to make clear instructions. If these are still too hard to follow, I'll make it so that just `pyenv init -` works, too, as previously, making `--path` optional -- but that will yield poor results in some setups (e.g. in distros that add `~/.local/bin` to `PATH` in profile) where it's preferreable to add a `PATH` entry at a different moment.**

### Tests
- [x] My PR adds the following unit tests (if any)

N/A

### CODE REVIEW

**Comment by shniubobo** [comment]
Please note that WSL also runs shells as login shells by default.

```console
$ [[ -o login ]] && echo "login"
login
$ zsh
$ [[ -o login ]] && echo "login"  # no output
```

Tested with Windows Terminal 1.10.2383.0 and:
1. Ubuntu 18.04 WSL, zsh 5.4.2
2. ArchWSL, zsh 5.8

Maybe the config for MacOS also applies to WSL?

**Comment by native-api** [comment]
> 
> 
> Please note that WSL also runs shells as login shells by default.
> 
> ```
> $ [[ -o login ]] && echo "login"
> login
> $ zsh
> $ [[ -o login ]] && echo "login"  # no output
> ```
> 
> Tested with Windows Terminal 1.10.2383.0 and:
> 
>     1. Ubuntu 18.04 WSL, zsh 5.4.2
> 
>     2. ArchWSL, zsh 5.8
> 
> 
> Maybe the config for MacOS also applies to WSL?

AFAICS, WSL is effectively an Ubuntu console session in a Windows terminal window. So instructions for Ubuntu should also apply to it, no need for a separate section.

**Comment by shniubobo** [comment]
> AFAICS, WSL is effectively an Ubuntu console session in a Windows terminal window. So instructions for Ubuntu should also apply to it, no need for a separate section.

Of course instructions for Ubuntu work for WSL, and of course no extra section is needed.

But instructions for Ubuntu also work for MacOS (except that an extra note is needed to tell users to run the shell as a login shell). Why should there be a seperate section for MacOS? Because there is no need to add three extra lines into `.profile` [1]. This is the same as WSL.

What I meant by saying "maybe the config for MacOS also applies to WSL" was that, instead of writing:

> MacOS, if Pyenv is installed with a Git checkout:
> In MacOS, restarting terminal windows is enough

we could write:

> MacOS or WSL, if Pyenv is installed with a Git checkout:
> In MacOS and WSL, restarting terminal windows is enough

[1] I have never used MacOS, but I assume lines in `.profile` are ignored if zsh is the login shell, as they should be on other OSes, and thus these extra lines will not break pyenv; please let me know if I am wrong.

**Comment by native-api** [comment]
> What I meant by saying "maybe the config for MacOS also applies to WSL" was that, instead of writing:
> 
> > MacOS, if Pyenv is installed with a Git checkout:
> > In MacOS, restarting terminal windows is enough
> 
> we could write:
> 
> > MacOS or WSL, if Pyenv is installed with a Git checkout:
> > In MacOS and WSL, restarting terminal windows is enough

That'd not be correct. As you can see, WSL is not a separate OS, it's merely another way to use a pre-existing Linux distro -- one of at least a few different ones as you've demonstrated yourself. So there isn't a single, separate environment called "WSL", and as such, there cannot be a separate, not to mention single, set of instructions for it, the instructions for whatever distro you're using through it apply instead.

Moreover, we do not officially support WSL (in particulae, do not test things on it as part of our maintenance) so I'm hesitant to treat it as a first-class citizen in the README as it would give users a false impression.

**Comment by anton-petrov** [comment]
I'll add some instructions for WSL2 later (using it in my work too)...
